gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/JDBCKnoxShellTableBuilder.java

```diff
@@ -109,12 +107,15 @@ public KnoxShellTable sql(String sql) throws IOException, SQLException {
     return this.table;
   }
 
-  private Connection createConnection() throws SQLException {
-    if (StringUtils.isNotBlank(username) && pass != null) {
-      return DriverManager.getConnection(connectionUrl, username, pass);
-    } else {
-      return DriverManager.getConnection(connectionUrl);
+  public Connection createConnection() throws SQLException {
+    Connection con = null;
+    if (username != null && pass != null) {
+      con = DriverManager.getConnection(connectionUrl, username, pass);
+    }
+    else {
+      con = DriverManager.getConnection(connectionUrl);
     }
+    return con;
```

Comment:
I think the older version checked empty string and this change removes that check on the username?

Reply:
hmmm, I wonder why it only had it for username and not pass.
Maybe an empty password is valid?
I don't think that I wrote that but that sort of makes sense.
I'll change it.

Reply:
Yea @smolnar82 wrote it not too long ago. I had commented that empty pass MAY be valid where empty username definitely isn't valid.

Reply:
The `public` visibility is an issue with how AspectJ is configured as of now. If this goes in, I need to figure out how to make it work. I haven't checked it yet, but package-private would not be enough?

Reply:
package protected will not be sufficient since it is required to called as part of the fluent API for "applications" that want to own the management of the connection rather than rely on it to be opened and closed within the call itself.

See this sort of code where the connection is externalized:

  protected Connection getConnection(KnoxDataSource ds, String user, String pass) throws SQLException, Exception {
    Connection conn = getConnectionFromSession(ds);
    if (conn == null) {
      if (user != null && pass != null) {
        conn = KnoxShellTable.builder().jdbc()
          .connectTo(ds.getConnectStr())
          .driver(ds.getDriver())
          .username(user)
          .password(pass)
          .createConnection();
      }
      else {
        conn = KnoxShellTable.builder().jdbc()
          .connectTo(ds.getConnectStr())
          .driver(ds.getDriver())
          .createConnection();
      }
      HashMap<String, Connection> connections =
          (HashMap<String, Connection>) getVariables()
          .getOrDefault(KNOXDATASOURCE_CONNECTIONS,
              new HashMap<String, Connection>());
      connections.put(ds.getName(), conn);
      getVariables().put(KNOXDATASOURCE_CONNECTIONS, connections);
    }
    return conn;
  }

Reply:
It is also noting that this configuration of aspectj that limits anyone adding new public methods is too brittle and needs to be addressed if possible.

Reply:
I guess we could completely externalize the creation of the connection if it is going to be externally managed but that still doesn't address the brittleness aspectj is imposing on the builders.